PowerEditor/src/ScitillaComponent/AutoCompletion.cpp

```diff
@@ -89,7 +88,16 @@ bool AutoCompletion::showApiAndWordComplete()
 	bool canStop = false;
 	for (size_t i = 0, kwlen = _keyWordArray.size(); i < kwlen; ++i)
 	{
-		if (_keyWordArray[i].compare(0, len, beginChars) == 0)
+
+		generic_string keytemp;
+		std::transform(_keyWordArray[i].begin(), _keyWordArray[i].end(), std::back_inserter(keytemp), ::tolower);
+		generic_string expr(beginChars);
+		generic_string beginchartmp;
+		std::transform(expr.begin(), expr.end(), std::back_inserter(beginchartmp), ::tolower);
+
+		//if (_keyWordArray[i].compare(0, len, beginChars) == 0)
+		//if (keytemp.compare(0, len, beginchartmp) == 0)
+		if (((keytemp.compare(0, len, beginchartmp) == 0) && _ignoreCase) || (_keyWordArray[i].compare(0, len, beginChars) == 0))

```

Comment:
This is inefficient because it always transforms the two strings even if `_ignoreCase` is false.